dnsutil/util_test.go

```diff
@@ -0,0 +1,110 @@
+package dnsutil
+
+import "testing"
+
+// func AddDomain
+
+func TestAddDomain_nodot(t *testing.T) {
+	var tests = []struct{ experiment, expected string }{
+		{"@", "example.com"},
+		{"foo", "foo.example.com"},
+		{"foo.", "foo."}, // Not really "nodot" but this is expected behavior.
+	}
+	for _, test := range tests {
+		actual := AddDomain("example.com", test.experiment)
+		if test.expected != actual {
+			t.Errorf("%v: expected (%v) got (%v)\n", test.experiment, test.expected, actual)
+		}
+	}
+}
+
+func TestAddDomain_dot(t *testing.T) {
+	var tests = []struct{ experiment, expected string }{
+		{"@", "example.com."},
+		{"foo", "foo.example.com."},
+		{"foo.", "foo."},
+	}
+	for _, test := range tests {
+		actual := AddDomain("example.com.", test.experiment)
+		if test.expected != actual {
+			t.Errorf("%v: expected (%v) got (%v)\n", test.experiment, test.expected, actual)
+		}
+	}
+}
+
+func TestAddDomain_oddball(t *testing.T) {
+	// In general origin should not be "" or "." but at least
+	// these tests verify we don't crash and will keep results
+	// from changing unexpectedly.
+	var tests = []struct{ e1, e2, expected string }{
+		{"", "*.", "*."},
+		{"", "@", "@"},
+		{"", "foobar", "foobar"},
+		{"", "foobar.", "foobar."},
+		{".", "*.", "*."},
+		{".", "@", "@"},
+		{".", "foobar", "foobar."},
+		{".", "foobar.", "foobar."},
+	}
+	for _, test := range tests {
+		actual := AddDomain(test.e1, test.e2)
+		if test.expected != actual {
+			t.Errorf("AddDomain(%#v, %#v): expected (%v) got (%v)\n", test.e1, test.e2, test.expected, actual)
+		}
+	}
+}
+
+// func TrimDomain
+
+func TestStripTrimDomain_normal(t *testing.T) {
+	var tests = []struct{ experiment, expected string }{
+		{"", "@"},
+		{".", "."},
+		{"f.", "f."},
+		{".f.", ".f."},
+		{"*.example.com", "*"},
+		{".foo.example.com.", ".foo"},
+		{".foo.example.com", ".foo"},
+		{".foo.ronco.com.", ".foo.ronco.com."},
+		{".foo.ronco.com", ".foo.ronco.com"},
+		{"example.com", "@"},
+		{"example.com.", "@"},
+		{"foo.", "foo."},
+		{"foo.example.com", "foo"},
+		{"foo.example.com.", "foo"},
+		{"serverfault.com", "serverfault.com"},
+		{"serverfault.com.", "serverfault.com."},
+	}
+	for _, dom := range []string{"example.com", "example.com."} {
+		for _, test := range tests {
+			actual := TrimDomain(dom, test.experiment)
+			if test.expected != actual {
+				t.Errorf("TrimDomain(%#v, %#v): expected (%v) got (%v)\n", dom, test.experiment, test.expected, actual)
+			}
+		}
+	}
+}
+
+func TestTrimDomain_shortdomain(t *testing.T) {
+	// This test shouldn't be needed but I was weary of off-by-one errors
+	// due to a very short TLD.
+	var tests = []struct{ experiment, expected string }{
```

Comment:
move tests that are actually different into TestStripDomainNameNormal (and drop Normal)